.github/workflows/test_on_push.yml

```diff
@@ -39,7 +39,6 @@ jobs:
         os: [ubuntu-latest, macos-12, windows-latest]
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         # We check coverage on Ubuntu with Python 3.11, so we skip unit tests for it here
-        # TODO: check coverage with Python 3.12 when [odes] supports it
         exclude:
           - os: ubuntu-latest
             python-version: "3.11"
```

Comment:
We should probably remove this. If we change the coverage test, and forget this line then we skip an architecture

Reply:
Could you bump the coverage job to Python 3.12? It will get us some good [speedup](https://nedbatchelder.com/blog/202312/coveragepy_with_sysmonitoring.html) for it (the article is a bit dated so the feature is a bit more stable now)